datacapture/src/main/java/com/google/android/fhir/datacapture/validation/ConstraintValidator.kt

```diff
@@ -34,5 +34,9 @@ internal interface ConstraintValidator {
     questionnaireResponseItem: QuestionnaireResponse.QuestionnaireResponseItemComponent
   ): ConstraintValidationResult
 
-  data class ConstraintValidationResult(val isValid: Boolean, val message: String?)
+  data class ConstraintValidationResult(
+    val isValid: Boolean,
+    val linkId: String?,
```

Comment:
can this be mandatory?

```suggestion
    val linkId: String,
```